x-pack/plugins/data_enhanced/public/search/ui/connected_search_session_indicator/connected_search_session_indicator.tsx

```diff
@@ -4,33 +4,47 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
-import { debounceTime, distinctUntilChanged, map } from 'rxjs/operators';
+import React, { useRef } from 'react';
+import { debounce, distinctUntilChanged, map } from 'rxjs/operators';
+import { timer } from 'rxjs';
 import useObservable from 'react-use/lib/useObservable';
 import { i18n } from '@kbn/i18n';
-import { SearchSessionIndicator } from '../search_session_indicator';
-import { ISessionService, TimefilterContract } from '../../../../../../../src/plugins/data/public/';
+import { SearchSessionIndicator, SearchSessionIndicatorRef } from '../search_session_indicator';
+import {
+  ISessionService,
+  SearchSessionState,
+  TimefilterContract,
+} from '../../../../../../../src/plugins/data/public/';
 import { RedirectAppLinks } from '../../../../../../../src/plugins/kibana_react/public';
 import { ApplicationStart } from '../../../../../../../src/core/public';
+import { IStorageWrapper } from '../../../../../../../src/plugins/kibana_utils/public';
+import { useSearchSessionTour } from './search_session_tour';
 
 export interface SearchSessionIndicatorDeps {
   sessionService: ISessionService;
   timeFilter: TimefilterContract;
   application: ApplicationStart;
+  storage: IStorageWrapper;
 }
 
 export const createConnectedSearchSessionIndicator = ({
   sessionService,
   application,
   timeFilter,
+  storage,
 }: SearchSessionIndicatorDeps): React.FC => {
   const isAutoRefreshEnabled = () => !timeFilter.getRefreshInterval().pause;
   const isAutoRefreshEnabled$ = timeFilter
     .getRefreshIntervalUpdate$()
     .pipe(map(isAutoRefreshEnabled), distinctUntilChanged());
 
+  const debouncedSessionServiceState$ = sessionService.state$.pipe(
```

Comment:
Couldn't you listen to the `appId$` change instead?

Reply:
No, for example: navigating from a dashboard page to a dashboard listing page